app-arch/libarchive/libarchive-3.7.9.recipe

```diff
@@ -55,27 +52,48 @@ REQUIRES="
 	lib:libcrypto$secondaryArchSuffix
 	lib:libiconv$secondaryArchSuffix
 	lib:liblzma$secondaryArchSuffix
-	lib:libssl$secondaryArchSuffix
 	lib:libxml2$secondaryArchSuffix
 	lib:libz$secondaryArchSuffix
+	lib:libzstd$secondaryArchSuffix
 	"
 
 PROVIDES_devel="
-	libarchive${secondaryArchSuffix}_devel = $portVersion compat >= 3.0
+	libarchive${secondaryArchSuffix}_devel = $portVersionCompat
 	devel:libarchive$secondaryArchSuffix = $libVersionCompat
 	"
 REQUIRES_devel="
 	libarchive$secondaryArchSuffix == $portVersion base
+	devel:iconv$secondaryArchSuffix
+	devel:libcrypto$secondaryArchSuffix
+	"
+
+PROVIDES_tools="
+	libarchive${secondaryArchSuffix}_tools = $portVersionCompat
+	cmd:bsdcat$commandSuffix = $portVersionCompat
+	cmd:bsdcpio$commandSuffix = $portVersionCompat
+	cmd:bsdtar$commandSuffix = $portVersionCompat
+	cmd:bsdunzip$commandSuffix = $portVersionCompat
+	"
+REQUIRES_tools="
+	libarchive$secondaryArchSuffix == $portVersion base
+	haiku$secondaryArchSuffix
+	lib:libbz2$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:liblzma$secondaryArchSuffix
+	lib:libxml2$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libzstd$secondaryArchSuffix
 	"
 
 BUILD_REQUIRES="
 	haiku${secondaryArchSuffix}_devel
 	devel:libbz2$secondaryArchSuffix
-	devel:libcrypto$secondaryArchSuffix >= 3
+	devel:libcrypto$secondaryArchSuffix
 	devel:liblzma$secondaryArchSuffix
-	devel:libssl$secondaryArchSuffix >= 3
 	devel:libxml2$secondaryArchSuffix
 	devel:libz$secondaryArchSuffix
+	devel:libzstd$secondaryArchSuffix
```

Comment:
is it required? adding this without a notice in the commit message isn't going to help...

Reply:
Looking at Gentoo and Fedora dependencies it's not mentioned as a build dependency?

https://packages.gentoo.org/packages/app-arch/libarchive/dependencies
https://src.fedoraproject.org/rpms/mingw-libarchive/blob/rawhide/f/mingw-libarchive.spec

Arch does mention it:

https://archlinux.org/packages/core/x86_64/libarchive/

Doesn't make it any clearer ;)

Reply:
I don't really like to add support for something that no one asks for, this is how we ended with a cmake qt dependency.

Reply:
what's disturbing is bumping a version, and adding a non-required dependency with a title "update to version 3.7.9".
make it "update to version 3.7.9, add zstd dependency" and that's clearer.

I don't care for the completeness sake honestly. enough on my plates.

Reply:
Main issue for updating this was the missing devel mentions for libiconv/libcrypto, aside from that, this is mentioned in the commit message bullet list?

Reply:
Looks (very) good to me.